mods/gen2/moves.js

```diff
@@ -12,13 +12,17 @@ exports.BattleMovedex = {
 			if (target.boosts.atk >= 6) {
 				return false;
 			}
+
+			var boost = 0;
+
 			if (target.hp <= target.maxhp / 2) {
-				this.boost({atk: 2});
-				return false;
+				boost = (target.boost ? target.boost.atk : 0) + 2;
```

Comment:
My understanding of Belly Drum in Gen 2 is this:

1. Fails if the user's Attack stat is already 999 (even if the user is at +5)
2. Do +2 Attack once, then clamp the Attack stat between 1 and 999
3. If the user's HP would drop to 0, fail then break; otherwise it loses 1/2 max HP
4. While the user's Attack stat stage is less than +6
	- if the Attack stat is 999, break
	- do +2 Attack, then clamp the Attack stat between 1 and 999

(Note: the game won't attempt to go beyond +6 Attack)

I just did a quick test on a Pokemon with 290 Attack; using Belly Drum at +0 (-> +6) requires 12 Growls to lower it to -6, but using Belly Drum at -1 (-> +5) requires only 11 Growls. Seems like that supports the above.

No message besides "But it failed!" is shown if Belly Drum fails at step 3, it just raises your Attack silently. I'm guessing we want the client to display the multiplier anyway?